libs/@local/hash-graph-sdk/python/graph_sdk/client/concurrent.py

```diff
@@ -255,3 +262,60 @@ async def query_entities(self, query: BaseFilter, options: Options) -> Subgraph:
         )
 
         return await self.inner.query_entities(request)
+
+    async def create_entity(
+        self,
+        properties: U,
+        *,
+        link: LinkData | None = None,
+        owned_by_id: UUID,
+    ) -> Entity[U]:
+        """Create an entity."""
+        actor = assert_not_none(self.actor)
+
+        request = CreateEntityRequest(
+            actor_id=RecordCreatedById(root=actor),
+            entity_type_id=VersionedURL(root=AnyUrl(properties.info.identifier)),
+            entity_uuid=None,
+            link_data=link,
+            owned_by_id=OwnedById(root=owned_by_id),
+            properties=EntityProperties.model_validate(
+                properties.model_dump(by_alias=True),
+            ),
+        )
+
+        metadata = await self.inner.create_entity(request)
+
+        # We cannot do `Entity[type(properties)]` because of mypy,
+        # by directly calling the magic method we can bypass this.
+        # Annoying? Yes. Necessary? Also yes.
+        properties_type: type[U] = type(properties)
+        entity_type = cast(type[Entity[U]], Entity.__class_getitem__(properties_type))
+
+        return entity_type.from_create(metadata, link, properties)
+
+    async def update_entity(
+        self,
+        entity: Entity[U],
```

Comment:
I am still a bit unsure about this one, we create the types dynamically, so I don't know if `U` will ever be of use, as you cannot statically refer to any entity type.